### PR TITLE
docs: Add instructions on uninstalling Teleport

### LINKED ISSUE #22862: Move uninstall instructions to docs
This guide is in discussions

https://github.com/gravitational/teleport/discussions/9920

Customers don't think it's official and ask us to add this to docs instead.

### PR DESCRIPTION
**Pull request by webvictim**
Converts https://github.com/gravitational/teleport/discussions/9920 into a proper documentation page

Fixes https://github.com/gravitational/teleport/issues/22862

### CODE REVIEW

**Review by r0mant — APPROVED** [review]
Overall lgtm but I wonder if it would make sense to structure the guide into separate parts for uninstalling/clearing data for server components and client components explicitly. I'd let @ptgott and @alexfornuto decide.

File: docs/pages/management/admin/uninstall-teleport.mdx
Comment (webvictim) [reply docs/pages/management/admin/uninstall-teleport.mdx:379]: I wondered about this too, but I was a little wary about us changing the locations in code (especially for things like Teleport Connect) and not updating the docs. It might be useful though.

File: docs/pages/management/admin/uninstall-teleport.mdx
Comment (webvictim) [reply docs/pages/management/admin/uninstall-teleport.mdx:131]: I'll let Paul/Alex weigh in on this as to whether they think it's worth calling out with another admonition or similar.

File: docs/pages/management/admin/uninstall-teleport.mdx
Comment (zmb3) [inline docs/pages/management/admin/uninstall-teleport.mdx:13]: I don't know if we should say _bare-metal_ here. That's an overloaded term and this usage of it is arguably inaccurate.

**Review by zmb3 — APPROVED** [review]

File: docs/pages/management/admin/uninstall-teleport.mdx
Comment (webvictim) [reply docs/pages/management/admin/uninstall-teleport.mdx:136]: I've added this just to be clearer.

File: docs/pages/management/admin/uninstall-teleport.mdx
Comment (webvictim) [reply docs/pages/management/admin/uninstall-teleport.mdx:421]: I've added some clarifications.

**Comment by webvictim** [comment]
The bot keeps removing my review requests. @ptgott and/or @alexfornuto - please review to make sure this fits appropriate with our documentation style.

File: docs/pages/management/admin/uninstall-teleport.mdx
Comment (ptgott) [inline docs/pages/management/admin/uninstall-teleport.mdx:502]:
```suggestion
Any Teleport Services will stop appearing in your Teleport Web UI or the output of `tsh ls` once their last heartbeat has timed out. This usually occurs within 10-15 minutes of stopping the Teleport process.
```

**Review by ptgott — APPROVED** [review]
Thanks for this!

- Should we link to this guide from `docs/pages/installation.mdx`?
- We should add a link to `docs/management/admin.mdx`

**Pull request merged**
